### DISCUSSION TITLE
expand Player#setResourcePack() to include forced boolean and message

### DISCUSSION THREAD

**Discussion by YouHaveTrouble** [PurpurMC/Purpur#431] [__Completed__] [answered]
## Is your feature request related to a problem?
  Currently forced resource packs can be sent only via packet, which is problematic, because players don't get resource pack invulnerability when you do that.

## Describe the solution you'd like.
 `Player#setResourcePack(String url, String hash, boolean forced, Component message)`

## Describe alternatives you've considered.
  Messing with packets.

**Comment by BillyGalbreath** [comment, marked as answer]
Added in build 1332

https://github.com/pl3xgaming/Purpur/commit/1105e102842d30468250641dc1cf445f1dc0dfb9
